plugins/modules/aci_fabric_wide_settings.py

```diff
@@ -0,0 +1,313 @@
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        disable_remote_ep_learning=dict(type="bool"),
+        enforce_subnet_check=dict(type="bool"),
+        enforce_epg_vlan_validation=dict(type="bool"),
+        enforce_domain_validation=dict(type="bool"),
+        spine_opflex_client_auth=dict(type="bool"),
+        leaf_opflex_client_auth=dict(type="bool"),
+        spine_ssl_opflex=dict(type="bool"),
+        leaf_ssl_opflex=dict(type="bool"),
+        ssl_opflex_tls_10=dict(type="bool"),
+        ssl_opflex_tls_11=dict(type="bool"),
+        ssl_opflex_tls_12=dict(type="bool"),
+        reallocate_gipo=dict(type="bool"),
+        restrict_infra_vlan_traffic=dict(type="bool"),
+        state=dict(type="str", default="present", choices=["present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_together=[["ssl_opflex_tls_10", "ssl_opflex_tls_11", "ssl_opflex_tls_12"]],
+    )
+
+    aci = ACIModule(module)
+
+    disable_remote_ep_learning = aci.boolean(module.params.get("disable_remote_ep_learning"))
+    enforce_subnet_check = aci.boolean(module.params.get("enforce_subnet_check"))
+    enforce_epg_vlan_validation = aci.boolean(module.params.get("enforce_epg_vlan_validation"))
+    enforce_domain_validation = aci.boolean(module.params.get("enforce_domain_validation"))
+    spine_opflex_client_auth = aci.boolean(module.params.get("spine_opflex_client_auth"))
+    leaf_opflex_client_auth = aci.boolean(module.params.get("leaf_opflex_client_auth"))
+    spine_ssl_opflex = aci.boolean(module.params.get("spine_ssl_opflex"))
+    leaf_ssl_opflex = aci.boolean(module.params.get("leaf_ssl_opflex"))
+    ssl_opflex_tls_10 = aci.boolean(module.params.get("ssl_opflex_tls_10"))
+    ssl_opflex_tls_11 = aci.boolean(module.params.get("ssl_opflex_tls_11"))
+    ssl_opflex_tls_12 = aci.boolean(module.params.get("ssl_opflex_tls_12"))
+    reallocate_gipo = aci.boolean(module.params.get("reallocate_gipo"))
+    restrict_infra_vlan_traffic = aci.boolean(module.params.get("restrict_infra_vlan_traffic"))
+    state = module.params.get("state")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="infraSetPol",
+            aci_rn="infra/settings",
+            module_object=None,
+            target_filter=None,
```

Comment:
When not provided the obj.get("target_filter") and obj.get("module_object") should return None so these lines are not needed.

Reply:
removed